api/instructor/instructorRouter.js

```diff
@@ -2,11 +2,49 @@ const express = require('express');
 const authRequired = require('../middleware/authRequired');
 const Instructors = require('./instructorModel');
 const router = express.Router();
+const Profiles = require('../profile/profileModel');
+const oktaClient = require('../../lib/oktaClient');
 const {
   getInstructorId,
   checkInstructorExist,
 } = require('./instructorMiddleware');
 
+/* Create a new Instructor profile */
+router.post('/register', (req, res) => {
+  if (!req.body) return res.sendStatus(400);
+  const newInstructor = {
+    profile: {
+      firstName: req.body.firstName,
+      lastName: req.body.lastName,
+      email: req.body.email,
+      login: req.body.email,
+    },
+    credentials: {
+      password: {
+        value: req.body.password,
+      },
+    },
+  }; // now sending to Okta
+  oktaClient
+    .createUser(newInstructor)
+    .then((instructor) => {
+      return Profiles.create({
+        email: instructor.profile.email,
+        name: instructor.profile.firstName + ' ' + instructor.profile.lastName,
+        okta_id: instructor.id,
+        role_id: 3,
```

Comment:
We'll want to add a `pending` column here and in our migration and seeds for profiles. For instructors, it will default to `true`, to be changed to `false` when their application has been accepted through the admin's dashboard.

For all other users, so in `userRouter.js`, we'll default to `false`, as their registration will simply be accepted by default.

Reply:
Added these changes!